### PR TITLE
Add baremetal testing and benchmarking integration in Jenkins using Lava

### PR DESCRIPTION
**Pull request by frdeso**

### CODE REVIEW

File: scripts/lttng-baremetal-tests/build-kernel.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -xe
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/build-kernel.sh]: Add -u to have the script fail on undefined variables.

File: scripts/lttng-baremetal-tests/build-modules.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+echo 'modules-built.txt does not exist'
+echo 'So we built them against the kernel'
+
+
+# Use all CPU cores
+NPROC=$(nproc)
+
+
+$SCP_COMMAND jenkins-lava@storage:$STORAGE_KERNEL_MODULE_SYMVERS $LINUX_PATH/Module.symvers
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/build-modules.sh]: Use the FQDN of storage and maybe put it in a variable.

File: scripts/lttng-baremetal-tests/build-kernel.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+make --directory=$LINUX_PATH -j$NPROC bzImage modules 
+make --directory=$LINUX_PATH INSTALL_MOD_PATH=$MODULES_INSTALL_FOLDER modules_install 
+cp $LINUX_PATH/arch/x86/boot/bzImage $DEPLOYDIR/$KERNEL_COMMIT_ID.bzImage
+cp $LINUX_PATH/.config $DEPLOYDIR/$KERNEL_COMMIT_ID.config
+
+tar -czf "$DEPLOYDIR/$BUILD_NAME.linux.modules.tar.gz" -C "$MODULES_INSTALL_FOLDER/" ./
+
+$SCP_COMMAND $DEPLOYDIR/$KERNEL_COMMIT_ID.bzImage jenkins-lava@storage:$STORAGE_KERNEL_IMAGE
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/build-kernel.sh]: All paths built with variables should be double quoted, you should use shellcheck on the bash scripts and fix any errors.

File: scripts/lttng-baremetal-tests/check-build-needs.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -x
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/check-build-needs.sh]: Missing -e and -u

File: scripts/lttng-baremetal-tests/check-build-needs.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+		git clone https://github.com/lttng/lttng-ci $LTTNG_CI_PATH
+		cp $LTTNG_CI_PATH/lava/kernel/vanilla/x86_64_server.config $LINUX_PATH/.config
+
+		make --directory=$LINUX_PATH olddefconfig
+		make --directory=$LINUX_PATH modules_prepare
+fi
+
+
+
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/check-build-needs.sh]: Such whitespace :)

File: scripts/lttng-baremetal-tests/lava-submit.py [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+        })
+    return command
+
+def get_benchmarks_cmd():
+    command = OrderedDict({
+        'command': 'lava_test_shell',
+        'parameters': {
+            'testdef_repos': [
+                {
+                    'git-repo': 'https://github.com/frdeso/lttng-ci.git',
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/lava-submit.py]: Should point to github.com/lttng/lttng-ci

File: scripts/lttng-baremetal-tests/run-benchmarks.sh [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+set +x
+python lttng-ci/scripts/lttng-baremetal-tests/lava-submit.py \
+															-t benchmarks \
+															-j $JOB_NAME \
+															-l $LAVA_FRDESO_TOKEN \
+															-k $STORAGE_KERNEL_IMAGE \
+															-km $STORAGE_LINUX_MODULES \
+															-lm $STORAGE_LTTNG_MODULES \
+															-tc $LTTNG_TOOLS_COMMIT_ID
+
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/run-benchmarks.sh]: Muuuch whitespace!

File: scripts/lttng-baremetal-tests/run-tests.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+echo $LTTNG_TOOLS_COMMIT_ID
+echo $LTTNG_UST_COMMIT_ID
+
+set +x
+python lttng-ci/scripts/lttng-baremetal-tests/lava-submit.py \
+															-t tests \
+															-j $JOB_NAME \
+															-l $LAVA_FRDESO_TOKEN \
+															-k $STORAGE_KERNEL_IMAGE \
+															-km $STORAGE_LINUX_MODULES \
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/run-tests.sh]: Any reason for this much whitespace?

File: jobs/lttng-baremetal-tests.yaml [hunk 9]
```diff
@@ -0,0 +1,218 @@
+- defaults:
+    name: frdeso-lttng-kernel-tests
```
Comment (mjeanson) [inline jobs/lttng-baremetal-tests.yaml]: Official jobs should not start with a username.

**Comment by frdeso** [comment]
Please see new version of the patch set.

**Pull request merged**